src/iuse.cpp

```diff
@@ -4121,201 +4112,155 @@ int iuse::siphon(player *p, item *it, bool, const tripoint& )
+    return toolweapon_off(p, it,
+        false,
+        rng(0, 10) - it->damage > 5 && !p->is_underwater(),
+        20, _("With a roar, the electric chainsaw leaps to life!"),
+        _("You flip the switch, but nothing happens."));
 }
 
-int iuse::elec_chainsaw_off(player *p, item *it, bool, const tripoint& )
+int iuse::cs_lajatang_off(player *p, item *it, bool, const tripoint& )
 {
-    p->moves -= 80;
-    if (rng(0, 10) - it->damage > 5 && it->charges > 0 && !p->is_underwater()) {
-        sounds::sound( p->pos(), 20,
-                      _("With a roar, the electric chainsaw leaps to life!"));
-        it->make("elec_chainsaw_on");
-        it->active = true;
-    } else {
-        p->add_msg_if_player(_("You flip the switch, but nothing happens."));
-    }
-    return it->type->charges_to_use();
+    return toolweapon_off(p, it,
+        false,
+        rng(0, 10) - it->damage > 5 && it->charges > 1 && !p->is_underwater(),
+        40, _("With a roar, the chainsaws leap to life!"),
+        _("You yank the cords, but nothing happens."));
 }
-int iuse::elec_chainsaw_on(player *p, item *it, bool t, const tripoint& )
+
+int iuse::carver_off(player *p, item *it, bool, const tripoint& )
 {
-    if (p->is_underwater()) {
-        p->add_msg_if_player(_("Your chainsaw gurgles in the water and stops."));
-        it->make("elec_chainsaw_off");
-        it->active = false;
-    } else if (t) { // Effects while simply on
-        if (one_in(15)) {
-            sounds::ambient_sound( p->pos(), 12, _("Your electric chainsaw rumbles."));
-        }
-    } else { // Toggling
-        p->add_msg_if_player(_("Your electric chainsaw dies."));
-        it->make("elec_chainsaw_off");
-        it->active = false;
-    }
-    return it->type->charges_to_use();
+    return toolweapon_off(p, it,
+        false,
+        true,
+        20, _("The electric carver's serrated blades start buzzing!"),
+        _("You pull the trigger, but nothing happens."));
 }
 
-int iuse::cs_lajatang_off(player *p, item *it, bool, const tripoint& )
+int iuse::trimmer_off(player *p, item *it, bool, const tripoint& )
 {
-    p->moves -= 80;
-    if (rng(0, 10) - it->damage > 5 && it->charges > 1) {
-        sounds::sound( p->pos(), 40,
-                      _("With a roar, the chainsaws leap to life!"));
-        it->make("cs_lajatang_on");
-        it->active = true;
-    } else {
-        p->add_msg_if_player(_("You yank the cords, but nothing happens."));
-    }
-    return it->type->charges_to_use();
+    return toolweapon_off(p, it,
+        false,
+        rng(0, 10) - it->damage > 3,
+        15, _("With a roar, the hedge trimmer leaps to life!"),
+        _("You yank the cord, but nothing happens."));
 }
 
-int iuse::cs_lajatang_on(player *p, item *it, bool t, const tripoint& )
+int toolweapon_on(player *p, item *it, bool t, const char *tname, bool works_underwater, int sound_chance, int volume, char *sound, bool double_charge_cost = false)
```

Comment:
The `char*` arguments have to be `const`, otherwise it won't compile in release builds.